### DISCUSSION TITLE
Render token in body request instead of header

### DISCUSSION THREAD

**Discussion by Segrelove** [Gokul595/api_guard#39] [Q&A] [answered]
Hello Gokul! 

I just implemented API Guard for a nuxt project. Really easy and straightforward, which is really nice. 

But I was wondering how you can send the token on user authentication in the body request instead of the header ? 

As I said, I'm using the auth module of nuxt, and to store the token, they grab it from the body and not from the header.

I tweaked a bit the AuthenticationController to make it work. That's what I came up with : 

`module InternUsers
  class AuthenticationController < ApiGuard::AuthenticationController
    before_action :find_resource, only: [:create]
    before_action :authenticate_resource, only: [:destroy]

    def create
      if resource.authenticate(params[:password])
        create_token_and_set_header(resource, resource_name)
        render json: { token: headers["Access-Token"] } #<- I'm taking the generated token from the header to pass it in the body
      else
        render_error(422, message: I18n.t('api_guard.authentication.invalid_login_credentials'))
      end
    end
    
    ....
end`

**Comment by Gokul595** [comment, marked as answer]
@Segrelove Currently, there is no way to send the token in response body. The tweak that you added should work for now.

Thanks for bringing this. I will consider this and try to add an option in the gem to configure how the token should be sent in the response (body/header).
